connect/proxy/listener.go

```diff
@@ -166,9 +166,11 @@ func (l *Listener) Serve() error {
 
 // handleConn is the internal connection handler goroutine.
 func (l *Listener) handleConn(src net.Conn) {
-	defer src.Close()
-	// Make sure Listener.Close waits for this conn to be cleaned up.
-	defer l.connWG.Done()
```

Comment:
Defers are stack-based so `src.Close()` will be the last to be called in this function